### PR TITLE
Add API Gateway for Notification, Reply, and User Services

### PR DESCRIPTION
**Pull request by JinnySarita**
**Add API Gateway for Notification, Reply, and User Services**

Summary: This PR introduces an API gateway that serves as a centralized entry point for the noti-service, reply-service, and user-service. The implementation enhances service modularity and simplifies client interactions with the backend services.

### CODE REVIEW

**Review by sarunnut1236 — CHANGES_REQUESTED** [review]
ตั้งชื่อ PR กำกวมค้าบเตง

**Comment by JinnySarita** [comment]
> ตั้งชื่อ PR กำกวมค้าบเตง

แก้แล้วค้าบเตง

**Comment by JinnySarita** [comment]
> ตั้งชื่อ PR กำกวมค้าบเตง

แก้แล้วค้าบเตง

File: notification-service/src/controllers/notificationController.ts [hunk 1]
```diff
@@ -89,6 +94,7 @@ export const markNotificationsAsSeenByUserId = async (
   res: Response
 ): Promise<void> => {
   const userId = req.user?.userId;
+  console.log("mark!!")
```
Comment (sarunnut1236) [inline notification-service/src/controllers/notificationController.ts]: delete consolelog jraaa

**Review by sarunnut1236 — APPROVED** [review]
โค้ดใช้งานได้แล้ว โอเค ลุย!

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -3,9 +3,9 @@ name: Prettier
 on:
   push:
     branches: ["**"]
-  pull_request:
-    branches:
-      - main
+  # pull_request:
```
Comment (ThanatWongsamut) [inline .github/workflows/lint.yml:6]: Why we need to comment na @JinnySarita @sarunnut1236

File: api-gateway/src/routes/notification-route.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    `http://${process.env.NOTIFICATION_SERVICE_HOST}:${process.env.NOTIFICATION_SERVICE_PORT}` ||
+    "http://notification-service:5002";
+console.log("📢 NOTIFICATION_SERVICE_URL", NOTIFICATION_SERVICE_URL);
+
+notificationRouter.use(createLogMiddleware('notification-service'));
+
+notificationRouter.get('/health', async (req: Request, res: Response) => {
+    try {
+        // Forward the request to the notification service health check
+        console.log(`${NOTIFICATION_SERVICE_URL}/notifications/health-check`);
```
Comment (ThanatWongsamut) [inline api-gateway/src/routes/notification-route.ts]: Don't need to log

File: api-gateway/src/routes/notification-route.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        res.status(error.response?.status || 500).json({
+            message: "Error checking notification service health",
+        });
+    }
+})
+
+notificationRouter.get("/", async (req: Request, res: Response) => {
+    try {
+        // Forward the request to the getAllNotificationsByUserId
+        console.log(`${NOTIFICATION_SERVICE_URL}/notifications`);
```
Comment (ThanatWongsamut) [inline api-gateway/src/routes/notification-route.ts]: Same as before, Don't need to log

File: api-gateway/src/routes/notification-route.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        res.status(error.response?.status || 500).json({
+            message: "Error to getAllNotificationsByUserId from api-gateway",
+        });
+    }
+})
+
+notificationRouter.get("/unread", async (req: Request, res: Response) => {
+    try {
+        // Forward the request to the getAllNotificationsByUserId
+        console.log(`${NOTIFICATION_SERVICE_URL}/notifications/unread`);
```
Comment (ThanatWongsamut) [inline api-gateway/src/routes/notification-route.ts]: Same as before, Don't need to log

File: api-gateway/src/routes/notification-route.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        res.status(error.response?.status || 500).json({
+            message: "Error to getAllNotificationsByUserId from api-gateway",
+        });
+    }
+})
+
+notificationRouter.patch("/", async (req: Request, res: Response) => {
+    try {
+        // Forward the request to the markNotificationsAsSeenByUserId
+        console.log(`${NOTIFICATION_SERVICE_URL}/notifications`);
```
Comment (ThanatWongsamut) [inline api-gateway/src/routes/notification-route.ts]: Same as before, Don't need to log

File: api-gateway/src/routes/reply-route.ts [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    `http://${process.env.REPLY_SERVICE_HOST}:${process.env.REPLY_SERVICE_PORT}` ||
+    "http://reply-service:5003";
+console.log("💬 REPLY_SERVICE_URL", REPLY_SERVICE_URL);
+
+replyRouter.use(createLogMiddleware('reply-service'));
+
+replyRouter.get("/health", async (req: Request, res: Response) => {
+    try {
+        // Forward the request to the asset service health check
+        console.log(`${REPLY_SERVICE_URL}/replies/health-check`);
```
Comment (ThanatWongsamut) [inline api-gateway/src/routes/reply-route.ts]: Same as before, Don't need to log

File: notification-service/src/repositories/notificationRepository.ts [hunk 8]
```diff
@@ -53,9 +53,6 @@ export const markNotificationsAsSeenByUserIdRepo = async (
       { isSeen: true }
     );
 
-    if (result.modifiedCount === 0) {
```
Comment (ThanatWongsamut) [inline notification-service/src/repositories/notificationRepository.ts:56]: Why we need to remove this

**Review by ThanatWongsamut — CHANGES_REQUESTED** [review]
Left some comment

File: .github/workflows/lint.yml (hunk 2 above)
Comment (sarunnut1236) [reply .github/workflows/lint.yml:6]: According to this code, it runs prettier twice, its redundant.

File: notification-service/src/controllers/notificationController.ts (hunk 1 above)
Comment (JinnySarita) [reply notification-service/src/controllers/notificationController.ts]: done kaa

File: api-gateway/src/routes/notification-route.ts (hunk 3 above)
Comment (JinnySarita) [reply api-gateway/src/routes/notification-route.ts]: done kaa

File: api-gateway/src/routes/notification-route.ts (hunk 4 above)
Comment (JinnySarita) [reply api-gateway/src/routes/notification-route.ts]: done kaa

File: api-gateway/src/routes/notification-route.ts (hunk 5 above)
Comment (JinnySarita) [reply api-gateway/src/routes/notification-route.ts]: done kaa

File: api-gateway/src/routes/notification-route.ts (hunk 6 above)
Comment (JinnySarita) [reply api-gateway/src/routes/notification-route.ts]: done kaa

File: api-gateway/src/routes/reply-route.ts (hunk 7 above)
Comment (JinnySarita) [reply api-gateway/src/routes/reply-route.ts]: done kaa

File: notification-service/src/repositories/notificationRepository.ts (hunk 8 above)
Comment (JinnySarita) [reply notification-service/src/repositories/notificationRepository.ts:56]: That code throws an error if no notifications are found to update, which results in a 500 error. But in reality, this isn’t an error—it just means there was nothing to update. Returning a 204 No Content makes more sense and avoids confusion, letting the client know that everything’s working as expected, even if there’s nothing new to update.

**Review by ThanatWongsamut — APPROVED** [review]
Nice

**Pull request merged**